flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/batch/connectors/hive/HiveInputFormatTest.java

```diff
@@ -122,4 +124,55 @@ public void testReadFromHiveInputFormat() throws Exception {
 		Assert.assertEquals("3,3,a,3000,3.33", rows.get(2).toString());
 		Assert.assertEquals("4,4,a,4000,4.44", rows.get(3).toString());
 	}
+
+	@Test
+	public void testReadComplextDataTypeFromHiveInputFormat() throws Exception {
+		final String dbName = "default";
+		final String tblName = "complext_test";
+
+		TableSchema.Builder builder = new TableSchema.Builder();
+		builder.fields(new String[]{"a", "m", "s"}, new DataType[]{
+				DataTypes.ARRAY(DataTypes.INT()),
+				DataTypes.MAP(DataTypes.INT(), DataTypes.STRING()),
+				DataTypes.ROW(DataTypes.FIELD("f1", DataTypes.INT()), DataTypes.FIELD("f2", DataTypes.STRING()))});
+
+		//Now we used metaStore client to create hive table instead of using hiveCatalog for it doesn't support set
+		//serDe temporarily.
```

Comment:
can you open a ticket for this work?

Reply:
ok, I created [FLINK-13061](https://issues.apache.org/jira/browse/FLINK-13061) to track it.